### ISSUE TITLE
19 System.IO.Pipes tests failing all Windows builds

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#1972]
19 of our System.IO.Pipes tests just started failing in every build, locally and in CI.  It appears something has changed regarding System.Threading.Overlapped, causing the following failure in each test:
```
System.TypeLoadException : Could not load type 'System.Threading.Overlapped' from assembly 'System.Threading.Overlapped, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```
cc: @ericstj, @davkean, @ericeil

**Comment by ericstj** [comment]
@ericeil removed the Overlapped type, I suspect you have a mix of old/new packages.  Let me see if I can repro locally.

**Comment by stephentoub** [comment]
> @ericeil removed the Overlapped type

It looks like not completely:
https://github.com/dotnet/corefx/blob/master/src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs#L278

> Let me see if I can repro locally.

Thanks.  Note that every Windows CI build is failing because of this, so it's not just on my machine, e.g.
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug_prtest/1657/console
This is why all of the recent PRs have failed.

**Comment by stephentoub** [comment]
So it looks like the System.Threading.Overlapped 4.0.0-beta-22927 package is messed up.  The ref assembly contains Overlapped, which is why Pipes compiles even though NamedPipeServerStream uses it, but then the lib assembly in DNXCore50 is the newer surface area without Overlapped.

**Comment by ericstj** [comment]
It's not the package.  The package is consistent.  It's bringing in more than one version of the package.  I repro'ed and I see a test is pulling in 4.0.0-beta-23008.

**Comment by stephentoub** [comment]
> The package is consistent

You're right; the lib DNXCore50 assembly doesn't contain Overlapped because it's type forwarded to mscorlib.

**Comment by ericstj** [comment]
It's the CopyTestToTestDirectory target resolving the [test-runtime-project.json](https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/Targets/test-runtime-project.json) from buildtools.  It looks like @mmitche updated this to use * versions, but didn't check in a locked project.lock.json.  This causes nuget (dnu) to use the latest one it can find in any feed.  As such the implementation builds against 22927 but the tests run against 23008.

**Comment by ericstj** [comment]
That said, Pipes is still in a broken state.  It needs to be updated to use ThreadPoolBoundHandle.  @ericeil, I thought you were going to do this?

**Comment by stephentoub** [comment]
It looks like NamedPipeServerStream was just missed as part of Eric's change:
https://github.com/dotnet/corefx/commit/87270686b2e8d78953d1787422671faaf25eccb9

**Comment by mmitche** [comment]
@ericstj My changes to build tools haven't actually been moved into corefx yet.  Could it be something else?

**Comment by ericstj** [comment]
Its actually one of my changes that caused this: https://github.com/dotnet/buildtools/commit/03d53d6ce8cc36994d450d5a7e698b53beb1303f, in conjunction with the feed being updated, and corefx being "locked".  I am making a fix to have buildtools lock the dependencies.

**Comment by mmitche** [comment]
@ericstj Sounds good.  I also have one minor fix to my last commit that needs to go in to before corefx buildtools can be updated.  I'll PR it in a moment

**Comment by ericstj** [comment]
I noticed that and have a fix in my branch, either way should work.

**Comment by stephentoub** [comment]
Fixed with #1974
